rclpy/src/rclpy/_rclpy.c

```diff
@@ -973,7 +1068,7 @@ rclpy_get_validation_error_for_full_topic_name(PyObject * Py_UNUSED(self), PyObj
   int validation_result;
   size_t invalid_index;
   rmw_ret_t ret = rmw_validate_full_topic_name(topic_name, &validation_result, &invalid_index);
-  if (ret != RMW_RET_OK) {
+  if (ret != RCL_RET_OK) {
```

Comment:
This change (and the related ones) is not right, since the `ret` in this case is an `rmw_ret_t` which was returned from a `rmw_*` function, and so `RMW_RET_OK` is the right thing to use.

Reply:
Sorry for the global search and replace...I will fix it.

On the other hand, I noticed that `rcl_publisher_get_subscription_count()` returns a `rmw_ret_t`. Is that correct?

Reply:
Yeah... those look like bugs to me. `rcl_publisher_get_subscription_count()` should (and does) return rcl ret codes, and `rmw_serialized_message_init()` should technically be assigned into an `rmw_ret_t` and translated into an `rcl_ret_t`. There is a function for that:

https://github.com/ros2/rcl/blob/76b0d5dc34cdd7505f40668210718471c7926850/rcl/src/rcl/common.h#L27

But it is unfortunately not public right now.

Reply:
https://github.com/ros2/rcl/pull/559